src/builtins/alias.c

```diff
@@ -72,13 +74,13 @@ int builtin_alias(int argc, char **argv, psh_state *state)
 
     xfree(argstr);
 
-    //if (strcmp(alias, check_for_alias(alias)) == 0)
-    //{
-    //    add_alias(alias, value);
-    //} else
-    //{
-    //    OUT2E("Alias %s already exists with %s\n", alias, check_for_alias(alias));
-    //}
+    if (check_for_alias(state->alias_table, alias) == NULL)
+    {
+        add_alias(state->alias_table, alias, value);
+    } else
+    {
+        OUT2E("Alias %s already exists with %s\n", alias, check_for_alias(state->alias_table, alias));
+    }
```

Comment:
If `check_for_alias(state->alias_table, alias) != NULL` then some trash gets printed.
I tried to figure this out for over 1 hour but everything I try result in some trash output exept I put `puts(psh_hash_get(table, alias));` at line 34 then the correct output is printed.
I assume there is a problem with the memory in another function that causes this.